fmriprep/interfaces/freesurfer.py

```diff
@@ -190,6 +192,46 @@ def _run_interface(self, runtime):
         return runtime
 
 
+class TruncateLTA(object):
+    # Use a tuple in case some object produces multiple transforms
+    lta_outputs = ('out_lta_file',)
+
+    def _post_run_hook(self, runtime):
+        print('ENTERING post_run Truncatae RRR')
+
+        outputs = self.aggregate_outputs(runtime=runtime)
+        print(outputs)
+        outputs = self._list_outputs()
+
+        for lta_name in self.lta_outputs:
+            lta_file = outputs[lta_name]
+            if not isdefined(lta_file):
+                continue
+
+            print('CHANGING FILE')
+
+            with open(outputs['out_lta_file'], 'r') as f:
+                lines = f.readlines()
+
+            fixed = False
+            newfile = []
+
+            for line in lines:
+                if line.startswith('filename = ') and len(line.strip("\n")) >= 255:
+                    fixed = True
+                    newfile.append('filename = path_too_long\n')
+                else:
+                    newfile.append(line)
+
+            if fixed:
+                with open(outputs['out_lta_file'], 'w') as f:
+                    f.write(''.join(newfile))
+
+        runtime = super(TruncateLTA, self)._post_run_hook(runtime)
```

Comment:
Given that we don't need to do this before other things, I would prefer to let other mixins take precedence, so I would place this as the first line of the method.

Reply:
I am not used to mixing, so debugin this code, make me realize the following.

Just to be clear I resume the code I refer to :

class PatchedBBRegisterRPT(TruncateLTA, BBRegisterRPT):
   pass

both classs TruncateLTA and BBRegisterRPT have their own definition of _post_run_hook method. So when object PatchedBBRegisterRPT call this function the order of mixins make it use the first one that is the one define in TruncateLTA.

ok that is fine, but it unintuitive is the meaning of the call (define in TruncateLTA)

runtime = super(TruncateLTA, self)._post_run_hook(runtime)

At a first look it is not define since TruncateLTA is heriting from object where not method post_run_hook is define. In fact this method is calling the BBRegisterRPT._post_run_hook
I understand it like if the definition 
class PatchedBBRegisterRPT(TruncateLTA, BBRegisterRPT):
implyt that TruncateLTA herit from BBRegisterRPT

IS that correct ?

Any way I am sure that this "super._post_run_hook call in TruncateLTA is interpreted as BBRegisterRPT._post__run_hook
So if you place it at the begining, this will execute the mri_vol2vol function before we modify the filename path, and I get the segmentation fault again

So it has to be at the end